src/lib/PostgresMetaTriggers.ts

```diff
@@ -0,0 +1,295 @@
+import format, { ident, literal } from 'pg-format'
+import { triggersSql } from './sql'
+import { PostgresMetaResult, PostgresTrigger } from './types'
+import PostgresMetaSchemas from './PostgresMetaSchemas'
+
+export default class PostgresMetaTriggers {
+  query: (sql: string) => Promise<PostgresMetaResult<any>>
+
+  constructor(query: (sql: string) => Promise<PostgresMetaResult<any>>) {
+    this.query = query
+  }
+
+  async list(): Promise<PostgresMetaResult<PostgresTrigger[]>> {
+    return await this.query(enrichedTriggersSql)
+  }
+
+  async retrieve({ id }: { id: number }): Promise<PostgresMetaResult<PostgresTrigger[]>>
```

Comment:
So the reason I made this `PostgresTrigger[]` is because of something you caught and mentioned in one of the comments:

> we can have rows with duplicate ids

Wasn't sure at the time how I wanted to handle it but after speaking with @MildTomato I've made the change to return a single record where `events` is an array (e.g. ['INSERT', 'UPDATE]).